try to build functorch in CI, not just for functorch tests

bdhirsh opened the pull request:
@zou3519 I'll see how this goes (what are the odds that it just works 😃)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82602
* #82771
* #82601
* __->__ #82770
* #83265

bdhirsh committed:
try to build functorch in CI, not just for functorch tests

[ghstack-poisoned]

bdhirsh committed:
Update on "try to build functorch in CI, not just for functorch tests"

@zou3519 I'll see how this goes (what are the odds that it just works 😃)




[ghstack-poisoned]

zou3519 commented on .jenkins/pytorch/win-test-helpers/build_pytorch.bat:
```diff
@@ -154,6 +154,10 @@ python setup.py install --cmake && sccache --show-stats && (
     copy /Y "build\.ninja_log" "%PYTORCH_FINAL_PACKAGE_DIR%\"
   )
 )
+:: Build functorch on windows
+pushd functorch
+time python setup.py install
+popd
```
Windows is a little annoying because we don't transfer docker images between the build job and the test job. What happens is that in the build job, we literally upload the contents of the installed pytorch folder somewhere and then we download the contents during the test job.

Because of how complicated this is I think it's fine if we say this is not going to work on windows. How important is it to get test signal for this on windows?

bdhirsh replied:
We haven't had a (real) windows-only issue with the functionalization tests so far, so this seems ok to me.

zou3519 commented on .jenkins/pytorch/build.sh:
```diff
@@ -293,6 +293,8 @@ else
     pushd ../cpp-build/caffe2
     WERROR=1 VERBOSE=1 DEBUG=1 python "$BUILD_LIBTORCH_PY"
     popd
+    echo "Building functorch"
+    install_functorch
```
We can delete https://github.com/pytorch/pytorch/blob/bbe8d019f280478dc3b143f6988e3e5668499f28/.jenkins/pytorch/test.sh#L679 after this, right?

bdhirsh replied:
I think you're right - I'll delete it and see if there's any fallout

bdhirsh commented:
hmm it looks like (at least) `[linux-focal-py3.7-gcc7 / test (functorch, 1, 1, linux.2xlarge)](https://github.com/pytorch/pytorch/pull/82770#logs)` tests are complaining about functorch not being available. Maybe `[.jenkins/pytorch/build.sh](https://github.com/pytorch/pytorch/pull/82770/files/ef88d0f306ceb753229d847500945e93d0e5fb6f#diff-b37b0309da87441d54c9aefd44684281bdb2a8cb9ab7075e2e3290c282592d17)` doesn't cover all of the linux build/test jobs.

bdhirsh committed:
Update on "try to build functorch in CI, not just for functorch tests"

zou3519 I'll see how this goes (what are the odds that it just works 😃)




[ghstack-poisoned]
